modules/odo-getting-started-deploy-dot-net.adoc

```diff
@@ -0,0 +1,131 @@
+// Module included in the following assemblies:
+//
+// * cli_reference/developer_cli_odo/getting-started-with-odo/odo-deploying-dot-net.adoc
+
+:_content-type: Procedure
+[id="odo-dot-net_{context}"]
+
+= Deploying your .NET application
+
+Use the `odo deploy` command to deploy your application.
+
+.Prerequisites
+* You have sample code for a .NET application.
+* You have logged in to your {product-title} cluster and created a project.
+* You have initialized your application with `odo init`.
+* You have started developing your application with `odo dev`.
+* You have logged in to your image registry.
+* You have configured the container build platform.
+
+.Procedure
+. *Containerize the application:* To deploy your application, you must containerize it to build and push it to a registry. Create the following Dockerfile in the same directory:
++
+.Sample Dockerfile
+[source, terminal]
+----
+FROM registry.access.redhat.com/ubi8/dotnet-60:6.0 as builder
+WORKDIR /opt/app-root/src
+COPY --chown=1001 . .
+RUN dotnet publish -c Release
+
+
+FROM registry.access.redhat.com/ubi8/dotnet-60:6.0
+EXPOSE 8080
+COPY --from=builder /opt/app-root/src/bin /opt/app-root/src/bin
+WORKDIR /opt/app-root/src/bin/Release/net6.0/publish
+CMD ["dotnet", "app.dll"]
+----
+
+. *Modify the Devfile:* Edit the `devfile.yaml` and add the appropriate deployment code:
++
+[WARNING]
+====
+When copying and pasting to `devfile.yaml`, verify that the lines you insert are correctly indented.
+====
+.. `odo deploy` uses Devfile schema 2.2.0. Change the schema to reflect the change:
++
+[source, yaml]
+----
+# Deploy "kind" ID's use schema 2.2.0+
+schemaVersion: 2.2.0
+----
+
+.. Add the `variables` section:
++
+[source, yaml]
+----
+# Add the following variables code anywhere in devfile.yaml
+# This MUST be a container registry you are able to access
+variables:
+  CONTAINER_IMAGE: quay.io/MYUSERNAME/dotnet-odo-example
+  RESOURCE_NAME: my-dotnet-app
+  CONTAINER_PORT: "8080"
+  DOMAIN_NAME: dotnet.example.com
+----
+
+include::snippets/odo-deploy-commands.adoc[]
+include::snippets/odo-deploy-resources.adoc[]
+include::snippets/odo-deploy-route.adoc[]
+
```

Comment:
Do you think it's possible to add the whole Devfile? I know we haven't done it for upstream, and I intend to do it asap, but can we do it here? I can help by providing you  with the end result(Devfile).